scriptmodules/helpers.sh

```diff
@@ -133,6 +133,33 @@ function editFile() {
     [[ -n "$choice" ]] && echo "$choice" >"$file"
 }
 
+## @fn inputBox()
+## @param title title of dialog
+## @param text default text
+## @param minchars minimum chars to accept
+## @brief Opens an inputbox dialog and echoes resulting text. Uses the OSK if installed.
+function inputBox() {
+    local title="$1"
+    local text="$2"
+    local minchars="$3"
+    [[ -z "$minchars" ]] && minchars=0
+    local params=(--backtitle "$__backtitle" --inputbox "$title")
+    local osk="$(rp_getInstallPath joy2key)/osk.py"
+
+    if [[ -f "$osk" ]]; then
+        params+=(--minchars "$minchars")
+        text=$(python3 "$osk" "${params[@]}" 2>&1 >/dev/tty)
```

Comment:
With the changes from https://github.com/RetroPie/RetroPie-Setup/pull/3638/commits/0b41d5f20b5907310548ec22456d7d16cdda0641, `$text` as an optional (last) parameter will work with the OSK. Current version will ignore it, so I think it's safe to change this to:
```
text=$(python3 "$osk" "${params[@]}" "$text" 2>&1 >/dev/tty)
```

Reply:
The current version throws an error with an additional parameter - so my plan was to get this in, and then you could adjust your PR to add the default parameter. With the current version I get

```
python3 /opt/retropie/admin/joy2key/osk.py --backtitle 'test' --inputbox test --minchars 4 "default text"
usage: osk.py [-h] --backtitle BACKTITLE --inputbox INPUTBOX [--minchars [MINCHARS]]
osk.py: error: unrecognized arguments: default text
```

Reply:
Regarding handling "cancel" - thanks. I missed that. I will adjust and we can retest. Appreciate your help.

Reply:
> With the current version I get.

Ugh, you're right - the extra arg is not working with the current version (damn you `argparse` !). OK, I'll redo the changes after this is merged - probably squash the `inputBox` addition and `osk` changes into a single commit.

Reply:
No worries. I'll fix this PR up later today and merge it and then you can rebase/rework your changes. Thank you :-)